ibc-clients/ics07-tendermint/types/src/consensus_state.rs

```diff
@@ -52,27 +53,26 @@ impl TryFrom<RawConsensusState> for ConsensusState {
     fn try_from(raw: RawConsensusState) -> Result<Self, Self::Error> {
```

Comment:
I experimented with using `DecodingError` as the type Error instead of `TendermintClientError`, while still keeping `DecodingError` as a variant within `TendermintClientError`. Everything works without any additional changes. If this holds true for other cases, I suggest we stick with `DecodingError` for all `TryFrom` conversions.

Reply:
I think this will require quite a bit of work to find all of the relevant `TryFrom` conversions, especially if we want to move towards all of the conversions actually returning `DecodingError`s: that will require a lot of refactoring. So I think this should be scoped into its own separate issue.

Reply:
Opened https://github.com/cosmos/ibc-rs/issues/1330 to capture this work.

Reply:
Makes sense. Sure. Let's tackle it as a separate PR.